STRETCH.md

```diff
@@ -3,21 +3,136 @@
+
+For example:
+
+- The game should be for two players;
+- Player 1 has the "X" symbol and starts;
+- Player 2 has the "O" symbol and goes second;
+- The grid size is 3x3;
+- The player with 3 in a row of their symbol, either vertically, horizontally, or diagonally, wins the game!
+
+> **NOTE**: Of course, these are just guidelines. Feel free to expand upon these rules, such as a custom grid size, custom symbols, or additional rules, such as playing for Eth.
+
+## Stretch Challenge 7: Payable mint, royalties, and withdrawing
+
+Often, NFT projects would like to make a little money themselves. So, they introduce minting costs and royalties to their project. This challenge aims to create an NFT smart contract that includes both!
+
+- The mint should cost at least .05 Eth per NFT;
+- The royalties should be configured for trading on Opensea;
+- The funds in the contract should be withdrawable;
+- Only the contract owner should be able to withdraw the funds.
+
+> **TIP**: Opensea has recently written a [blog post](https://opensea.io/blog/announcements/on-creator-fees/) on royalties for new tokens that they will be enforcing around December. You can find the GitHub repository [here](https://github.com/ProjectOpenSea/operator-filter-registry).
+
+## Stretch Challenge 8: Staking
+
+A keyword that you will often encounter is staking. Many NFT projects use staking to incentivize users to keep their NFTs instead of selling them and generate passive income by receiving tokens. In the context of NFTs, this often boils down to the following:
+
+- A user stakes their NFT(s);
+- Every staked NFT generates some tokens per x amount of time;
+- After a certain amount of time, users can get their NFTs back by claiming them;
+- The generated tokens can also be claimed and used for whatever intended utility the NFT project created.
+
+Create staking functionality for an already existing smart contract, or create a new NFT project that has staking functionality.
+
+> **TIP**: You can use the NFT smart contract from the [Backend Challenges](./backend/README.md).
+
+## Stretch Challenge 9: Gas Optimization
+
+Gas makes the Ethereum network [turing complete](https://academy.binance.com/en/glossary/turing-complete). However, as the network becomes more and more congested, the gas prices rise, increasing smart contract deployment cost and also cost for users that want to, for example, mint NFTs.
+
+There are a lot of techniques and applications, however, that you can use to optimize gas usage within Ethereum. This challenge aims to create an NFT smart contract that uses some form of gas optimization.
+
+- You should at least use the [ERC1155](https://ethereum.org/en/developers/docs/standards/tokens/erc-1155/), [ERC721A](https://www.erc721a.org/) or [ERC721Psi](https://github.com/estarriolvetch/ERC721Psi) standard.
+
+## Stretch Challenge 10: Whitelisting/Allowlisting with Merkle Trees
```

Comment:
Let's stay with the term "allowlist"... I think bigger companies might have restricted their vocabulary using some sort of inclusivity document...